metadata/component.go

```diff
@@ -171,6 +171,38 @@ func (m *manager) DeleteComponent(name string) error {
 	return nil
 }
 
+func (m *manager) GetComponentParams(component string) (param.Params, error) {
```

Comment:
`manager` has a large number of functions. As you change these, it's a good time to start extracting functionality into other structures.